onedal/spmd/linear_model/tests/test_linear_regression.py

```diff
@@ -0,0 +1,124 @@
+    from sklearn.metrics import mean_squared_error
+
+    def split_datasets(n_ranks, *arrays):
+        first = arrays[0]
+        n_samples = first.shape[1]
+
+        percentage = 1 / float(n_ranks)
+        block = int(n_samples * percentage)
+        assert int(block * n_ranks) <= n_samples
+
+        n_arrays = len(arrays)
+        results = ([],) * n_arrays
+
+        for b in range(n_ranks):
+            is_last = b + 1 == n_ranks
+            for a in range(n_arrays):
+                first = block * b
+
+                if is_last:
+                    last = n_samples
+                else:
+                    last = first + block
+
+                shard = arrays[a][:, first:last]
+                results[a].append(shard)
+
+        return results
+
+    def run_spmd_training(comm, queue, params, X, y):
+        rank, size = comm.Get_rank(), comm.Get_size()
+        Xs, ys = split_datasets(size, X, y)
+
+        model = LinearRegressionSpmd(**params)
+        model.fit(Xs[rank], ys[rank], queue)
+
+        coef = model.coef_
+
+        test = (rank + 1) % size 
+        Xt, yt = Xs[test], yt[test]
+
+        n_samples = Xt.shape[0]
+        yp = model.predict(Xt, queue)
+        error = mean_squared_error(yt, yp)
+
+        return (n_samples, coef, error)
+
+    def run_groundtruth(params, X, y):
+        model = LinearRegressionGtr(**params)
+        model.fit(X, y)
+
+        coef = model.coef_
+
+        yp = model.predict(X)
+        error = mean_squared_error(y, yp)
+
+        return (coef, error)
+
+    def run_on_dataset(comm, queue, params, X, y):
+        gtr_coef, gtr_err = run_groundtruth(params, X, y)
+        spmd_coef, spmd_err, n_samples = run_spmd_training(comm, queue, params, X, y)
+
+        assert_allclose(gtr_coef, spmd_coef)
+
+        mse = np.array(spmd_err, dtype = np.float64)
+        global_mse = np.zeros(comm.Get_size(), dtype = np.float64)
+        comm.Allgather([mse, MPI.MPI_DOUBLE], [global_mse, MPI.MPI_DOUBLE])
+
+        nss = np.array(n_samples, dtype = np.float64)
+        global_nss = np.zeros(comm.Get_size(), dtype = np.float64)
+        comm.Allgather([nss, MPI.MPI_DOUBLE], [global_nss, MPI.MPI_DOUBLE])
+
+        mse = np.sum(global_mse * global_nss) / np.sum(global_nss)
+
+        assert_close(mse, gtr_err)
+                
+    @pytest.mark.parametrize('queue', get_queues())
+    def test_diabetes(queue):
+        comm = MPI.COMM_WORLD
+        params = { 'fit_intercept' : True }
+        X, y = load_diabetes(return_X_y=True)
+        run_on_dataset(comm, queue, params, X, y)
+
+    @pytest.mark.parametrize('queue', get_queues())
+    def test_generated(queue):
+        comm = MPI.COMM_WORLD
+        rank, size = comm.Get_rank(), comm.Get_size()
+        n_samples, n_features = size * 10, size * 11
+
+        params = { 'fit_intercept' : True }
+        X, y = make_regression(n_samples = n_samples, 
+                               n_features = n_features)
```

Comment:
Add random_state for reproducibility

Reply:
Fixed by deleting. Not clear how to test this functionality